entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java

```diff
@@ -31,8 +33,8 @@ public FilterTypeExpression convert(
       final LiteralConstant constant,
       final RequestContext requestContext)
       throws ConversionException {
-    final String id = columnIdentifier.getColumnName();
-    final String subDocPath = getSubDocPathById(entityAttributeMapping, id, requestContext);
+    final String columnId = columnIdentifier.getColumnName();
+    final String subDocPath = getSubDocPath(columnId, requestContext);
 
     final IdentifierExpression identifierExpression = IdentifierExpression.of(subDocPath);
```

Comment:
Can be simplified after injecting org.hypertrace.entity.query.service.converter.identifier.IdentifierExpressionConverter. 
```suggestion
    final IdentifierExpression identifierExpression = identifierExpressionConverter.convert(columnIdentifier);
```

Reply:
Yes. That's a nice suggestion!